src/useActivities.ts

```diff
@@ -68,7 +68,13 @@ export const useActivities = (
     const anchor = moment(currentDateHash)
 
     while (allActivities[anchor.format(DATE_HASH_FORMAT)]) {
-      count++
+      const daysActivities = allActivities[anchor.format(DATE_HASH_FORMAT)]
+      if (
+        daysActivities &&
+        daysActivities.some((activity) => activity.status === Status.Done)
+      ) {
+        count++
+      }
```

Comment:
```suggestion
  const dayHasDoneActivity = allActivities[anchor.format(DATE_HASH_FORMAT)].some(({ status }) => status === Status.Done)
  if (!dayHasDoneActivity) break
  count++
```

Maybe something like that? I don't mind different names, but the `while` condition means `allActivities[anchor.format(DATE_HASH_FORMAT)` will always be truthy so there is no need to check.

The `break` fixes another bug, which is the streak doesn't reset! If you skip a day it should go back to zero.

Reply:
Ah, good spot! Though Typescript seems to still think `allActivities[anchor.format(DATE_HASH_FORMAT)` could be undefined in the while loop somehow 
![image](https://user-images.githubusercontent.com/23635298/79550963-f4448a80-8090-11ea-9f6b-1cc06e3ed1ae.png)

Reply:
I guess this is another example where ts can't figure out that it's defined without a variable? so options are to do what I originally did or override with `!` . Have just read a bit about this and it's not totally clear to me which is better but seems like there's not possibility the value is undefined so I'd lean towards `!` https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
I see! I've actually opened my mac now, and I did have something I didn't even save on the loop; but I still wasn't checking for "does it actually have any done" part.

Yes, that problem is Typescript needing a variable to understand whether something's been checking or not. It seems we can get around it by defining the variable before the while loop starts.

So maybe we end up with:

```ts
const consecutiveStreak = useMemo(() => {
    let count = 0
    const anchor = moment(currentDateHash)
    let daysActivities = allActivities[anchor.format(DATE_HASH_FORMAT)]

    while (daysActivities) {
      if (daysActivities.every((activity) => activity.status !== Status.Done)) {
        break
      }
      count++
      anchor.subtract(1, 'day')
      daysActivities = allActivities[anchor.format(DATE_HASH_FORMAT)]
    }

    return count
  }, [allActivities, currentDateHash])
```

Reply:
Sure!

Reply:
(will commit and merge)